cypress/support/commands.js

```diff
@@ -77,12 +83,82 @@ Cypress.Commands.add('syncUpCodeChanges', () => {
 
 Cypress.Commands.add('openStepConfigurationTab', (step, stepIndex) => {
   stepIndex = stepIndex ?? 0;
-  cy.get(`[data-testid="viz-step-${step}"]`).eq(0).click();
+  cy.get(`[data-testid="viz-step-${step}"]`).eq(stepIndex).click();
   cy.get('[data-testid="configurationTab"]').click();
 });
 
+Cypress.Commands.add('closeStepConfigurationTab', () => {
+  cy.get('.pf-c-button.pf-m-plain').click();
+});
+
 Cypress.Commands.add('checkCodeSpanLine', (firstSpan, secondSpan) => {
   cy.get('.code-editor').contains(firstSpan).parent()
     .should('contain.text', firstSpan)
     .and('contain.text', secondSpan)
 });
+
+Cypress.Commands.add('interactWithInputObject', (inputName, value) => {
+  if (value != null) {
+    cy.get(`input[name="${inputName}"]`).clear().type(value);
+  } else {
+    cy.get(`input[name="${inputName}"]`).click();
+  }
+});
+
+Cypress.Commands.add('dragAndDropFromCatalog', (source, target, targetIndex) => {
+  targetIndex = targetIndex ?? 0;
+  const dataTransfer = new DataTransfer();
+  cy.get('[data-testid="toolbar-step-catalog-btn"]').click();
+  cy.get('[data-testid="catalog-step-actions"]').click();
+  cy.get('#stepSearch').type(`${source}`);
+  cy.get(`[data-testid="catalog-step-${source}"]`).trigger('dragstart', {
+    dataTransfer,
+  });
+  cy.get(`[data-testid="viz-step-${target}"]`).eq(targetIndex).trigger('drop', {
+    dataTransfer,
+  });
+});
+
+// Blocked due to: https://github.com/KaotoIO/kaoto-ui/issues/1381
+// Cypress.Commands.add('insertBranch', (insertIndex) => {
```

Comment:
Hi @unsortedhashsets, in [this pull request](https://github.com/KaotoIO/kaoto-ui/pull/1452) we're temporarily dropping support for adding branches from the mini catalog whenever there's a custom step view (aka step extension) available for said step.

Part of the reason to do so it's because at the moment, the backend identifies the branch type by it's content, and for properly generate the content we would need to use the extension, notice that this may change in the future.

Reply:
Yes, thank you. I saw that solution. And I have already prepared commands (in that PR) to avoid canvas "branching". I am going to refactor previous E2Es with "add branch" actions tomorrow 😌